lib/mfg.c

```diff
@@ -216,23 +237,7 @@ int switchtec_security_config_get(struct switchtec_dev *dev,
 		return ret;
```

Comment:
Why not move this call into get_configs() where we make other config get MRPC calls?

Reply:
At the moment, function `get_configs()` only returns raw MRPC reply data. It does not parse the data in any way.

Not all callers need parsing the raw data (e.g., some callers only care if the bit `spi_core_clk_high` is set). 

Also in GEN4, the `otp_valid` field is determined by if `MRPC_SECURITY_CONFIG_GET_EXT` succeeds, so it's hard to merge the MRPC call to `get_configs()`.

Reply:
In the function switchtec_security_config_get(), we have a call to MRPC_SECURITY_CONFIG_GET_EXT to get the raw reply data, or a call to get_configs() which calls MRPC_SECURITY_CONFIG_GET_GEN5 or MRPC_SECURITY_CONFIG_GET to get the raw data. We can have the code of the three MRPC calls into the get_configs() and then avoid duplication of the below code section:
```             
parse_otp_settings(&state->otp, reply.valid);
state->otp_valid = true;
```
I can think of adding a pass out parameter to get_configs() to indicate if otp is valid (GEN5 or MRPC_SECURITY_CONFIG_GET_EXT works)

Reply:
Good idea. Code is now updated, please review.